pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -391,6 +391,22 @@ func (d *defaultDriver) newElasticsearchClient(
 	)
 }
 
+func (d *defaultDriver) elasticsearchClientProvider(
+	ctx context.Context,
+	state *reconcile.ResourcesState,
+	user esclient.BasicAuth,
+	v version.Version,
+	caCerts []*x509.Certificate,
+) func(existingEsClient esclient.Client) esclient.Client {
+	return func(existingEsClient esclient.Client) esclient.Client {
+		url := services.ElasticsearchURL(d.ES, state.CurrentPodsByPhase[corev1.PodRunning])
+		if existingEsClient != nil && existingEsClient.HasProperties(v, user, url, caCerts) {
+			return existingEsClient
+		}
+		return d.newElasticsearchClient(ctx, state, user, v, caCerts)
```

Comment:
Should we close the existing client not reused here?
```suggestion
		existingEsClient.Close()
		return d.newElasticsearchClient(ctx, state, user, v, caCerts)
```